City-of-Helsinki/drupal-helfi-rekry
# Uhf 5297 search results

# [UHF-5297] Job search: search results in React (https://helsinkisolutionoffice.atlassian.net/browse/UHF-5297)
Creates a list of results from Elastic index with pagination and sortin options.

## What was done
* Create new module for search app
* Update node modules
* Add env variables, settings and a paragraph type to Drupal 
* Modify index settings
* Develop React app to to list results

## How to install

* Make sure your instance is up and running on correct branch.
  * `git fetch && git checkout origin/UHF-5297-search-results`
  * `make fresh`

* Set up things you need: 
  * Env;
    * Run `make ps`. Note the port that your `drupal-helfi-rekry-elastic` container is running in.
    * Add ☝️ to your .env as the `ELASTIC_PROXY_URL` variable (`http:localhost:{port-elastic-runs-in}`)
    * Run `make up`
    * `make shell` into the container. Verify that your `ELASTIC_PROXY_URL` is set by running `printenv | grep ELASTIC_PROXY_URL`
  * Elastic:
    * **If** you have set up index before, it might be a good idea to delete it beforehand with `curl -X DELETE http://localhost:{port-elastic-runs-in}/job_listings`
    * Run `drush sapi-rt && drush sapi-i`
    * `curl -i http://localhost:{port-elastic-runs-in}/job_listings` to verify that the index is up and running (you should get a response with info about the cluster)

* Import configurations:
  * Run `drush deploy`

## How to test
 
* Create of modify a landing page
* Add a paragraph of type 'Job search' to content area
* View page

You should see the rendered app on the page. Test for these things: 

* Sort options in the top right corner of results sorts the results
  * Newest first should sort listings based on `published_at` value (Julkaisun alkaminen). You can check this value by opening the nodes.
  * Application closing should sort listings based of `unpublish_at` value (value displayed on the result card).
* Pagination works 

## Designers review



## Other PRs


* Link to other PR


[UHF-5297]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-5297?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
